src/createAnimatedComponent/createAnimatedComponent.tsx

```diff
@@ -281,21 +127,21 @@ export default function createAnimatedComponent(
       }
     }
 
+    componentDidMount() {
+      this._attachNativeEvents();
+      this._JSPropUpdater.addOnJSPropsChangeListener(this);
+      this._attachAnimatedStyles();
+      this._inlinePropUpdater.attachInlineProps(this, this._getViewInfo());
```

Comment:
I'm not a fan of the fact that this PR is supposed to split `createAnimatedComponent` into several files, but actually changes the logic slightly in some places - like here. Could you maybe split it into two separate PRs - one just does the split and the other changes things like this?

Reply:
Sure, see here for part one: [LINK](https://github.com/software-mansion/react-native-reanimated/pull/5053)